### ISSUE TITLE
Extension causes multiple tests executions when running a single test

### ISSUE THREAD

**Issue by kakins** [FortuneN/FineCodeCoverage#278]
### Installed product versions
- Visual Studio: 17.2.6
- This extension: 1.1.179
- .NET 6
- XUnit 2.4.1
- XUnit 2.4.5

### Description
Running a single test in VS Test explorer seems to run all tests in a file

### Steps to recreate
1. Have multiple test methods within a single class
2. Right click in the VS Test window and click either Run or Debug test
3. The selected test reports back pass/failed
4.  **But behind the scenes all tests are being run**, but not all are reported
5. This only appears to happen when running a test in the VS Test runner, not using `dotnet test`
6. When I disable the extension, the problem goes away

### Current behavior
The way I'm able to determine all of the tests are run is by observing the outcomes.  For instance, some tests of mine write to a database.  I am seeing data that is defined and isolated in a completely separate test written to the database.

I've double checked this by placing breakpoints to ensure the logic being tested is only executed once.  But the observable outcomes show that it is being run multiple times.

### Expected behavior
A test should only be run once.

### Side Notes
I will see if I can find some better ways to demonstrate the behavior.

**Comment by tonyhallett** [comment]
Duplicate of https://github.com/FortuneN/FineCodeCoverage/issues/267

Please use the option for ms code coverage. Although until the next release there is an issue with async coverage.

**Comment by kakins** [comment]
@tonyhallett Ah, I see.  The documentation is a little hard to follow on the main page, but now I see this section, under the MS Code Coverage section:

> The old coverage would wait until tests have finished before starting the coverage tool to re-run all tests. 

Which describes exactly what I'm seeing.  Thanks.

**Comment by kakins** [comment]
Actually can I re-open this issue and ask one more follow up question?  

Enabling the MS Code Coverage option seemed to fix the problem.  However, now I'm seeing a ton of results show up in the coverage report for libraries that shouldn't be included...  specifically `Humanizer`, `Moq` and `XUnit`.  I'm assuming I need to do some sort of filtering, but isn't there a way to say "Just cover my solution" and not any referenced packages.

![image](https://user-images.githubusercontent.com/14915247/179845489-f7d79840-84ae-4d26-9d13-9666e171c480.png)

![image](https://user-images.githubusercontent.com/14915247/179845653-fc0ab355-218a-4b78-bcf6-d1f37f16d425.png)

![image](https://user-images.githubusercontent.com/14915247/179845740-6e8b712b-b6d6-45a1-8be6-b99b529198a8.png)

**Comment by kakins** [comment]
@tonyhallett Any comment on my previous questions?  Is there any easy configuration so that I don't have to specify multiple filters?  I am using the VS options-based configuration, not project-level.

**Comment by tonyhallett** [comment]
Yes there is - IncludeReferencedProjetcs, but there is a bug.

Will let you know when the change is on the marketplace.

**Comment by tonyhallett** [comment]
Fixed.  Please update the extension and set IncludeReferencedProjects to true.

Is this the behaviour you are expecting ?

**Comment by kakins** [comment]
I don't think so.  Maybe I didn't make the last question clear enough.  When I run code coverage, third-party libraries are included in the coverage results.  Rather than add multiple specific filters to exclude those libraries, I want to run coverage for just my solution/project.  

I've tried setting `IncludeReferencedProjects` to true/false, no change really.  
![image](https://user-images.githubusercontent.com/14915247/180901975-e16238b2-7856-4f5a-9907-631dfdcdd398.png)

FYI, the other filters shown in that screenshot that say `String[] Array` are really empty, I was playing around with those settings but there are no values currently in those settings.

**Comment by tonyhallett** [comment]
I understand what you want to achieve and the fix that I put in last night does, for the solution that I was using, remove all assemblies other than those in my assembly.  That's not to say that that the logic is completely sound.

This is part of the code that determines the referenced projects

```
       private async Task<List<ReferencedProject>> GetReferencedProjectsFromDteAsync()
        {
            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
            var project = dte.Solution.Projects.Cast<Project>().FirstOrDefault(p =>
            {
                ThreadHelper.ThrowIfNotOnUIThread();
                //have to try here as unloaded projects will throw
                var projectFullName = "";
                try
                {
                    projectFullName = p.FullName;
                }
                catch { }
                return projectFullName == ProjectFile;
            });

            if (project == null)
            {
                return null;
            }

            var vsproject = project.Object as VSLangProj.VSProject;
            return vsproject.References.Cast<VSLangProj.Reference>().Where(r => r.SourceProject != null).Select(r =>
            {
                ThreadHelper.ThrowIfNotOnUIThread();
                var assemblyName = Path.GetFileNameWithoutExtension(r.Path);
                return new ReferencedProject(r.SourceProject.FullName, assemblyName);
            }).ToList();

        }

```

Within your-test-project\bin\Debug\net6.0\fine-code-coverage\coverage-tool-output you will find the generated runsettings.  What do you see for ModulePaths Include ?

**Comment by kakins** [comment]
Ok I disabled/re-enabled the extension just to make sure.  But I'm still getting the same results.  
![image](https://user-images.githubusercontent.com/14915247/181009512-70abfbcc-5a40-4c1b-ba79-63ac592442d0.png)


Here's the file contents you asked for:

```
<?xml version="1.0" encoding="utf-8"?>
<RunSettings>
  <RunConfiguration>
    <ResultsDirectory>D:\Source\Project\Project.UnitTests\bin\Debug\net6.0\fine-code-coverage\coverage-tool-output</ResultsDirectory>
    <TestAdaptersPaths>C:\Users\krisa\AppData\Local\FineCodeCoverage\msCodeCoverage\17.1.0\build\netstandard1.0</TestAdaptersPaths>
    <CollectSourceInformation>False</CollectSourceInformation>
  </RunConfiguration>
  <DataCollectionRunSettings>
    <DataCollectors>
      <DataCollector friendlyName="Code Coverage" enabled="True">
        <Configuration>
          <CodeCoverage>
            <ModulePaths>
              <Exclude>
                <ModulePath>D:\\Source\\Project\\Project.UnitTests\\bin\\Debug\\net6.0\\Project.UnitTests.dll</ModulePath>
              </Exclude>
              <Include>
                <ModulePath>.*\\Project.dll$</ModulePath>
                <ModulePath>[Project.*]</ModulePath>
              </Include>
            </ModulePaths>
            <Functions>
              <Exclude></Exclude>
              <Include></Include>
            </Functions>
            <Attributes>
              <Exclude></Exclude>
              <Include></Include>
            </Attributes>
            <Sources>
              <Exclude></Exclude>
              <Include></Include>
            </Sources>
            <CompanyNames>
              <Exclude></Exclude>
              <Include>
                <CompanyName>MyCompany</CompanyName>
              </Include>
            </CompanyNames>
            <PublicKeyTokens>
              <Exclude></Exclude>
              <Include></Include>
            </PublicKeyTokens>
            <UseVerifiableInstrumentation>True</UseVerifiableInstrumentation>
            <AllowLowIntegrityProcesses>True</AllowLowIntegrityProcesses>
            <CollectFromChildProcesses>True</CollectFromChildProcesses>
            <CollectAspDotNet>False</CollectAspDotNet>
          </CodeCoverage>
          <Format>Cobertura</Format>
          <FCCGenerated />
        </Configuration>
      </DataCollector>
    </DataCollectors>
  </DataCollectionRunSettings>
</RunSettings>
```

**Comment by tonyhallett** [comment]
> the other filters shown in that screenshot that say String[] Array are really empty, 

Can't be if CompanyNames has an entry.  I believe you have added `<ModulePath>[Project.*]</ModulePath>` too.

Can you remove and see what you get.

**Comment by kakins** [comment]
Whoops!  Didn't realize I had those in there still.  Yeah that seemed to work 🥳 Thanks so much for your help!
